packages/drivers/odsp-driver/src/prefetchLatestSnapshot.ts

```diff
@@ -106,6 +110,7 @@ export async function prefetchLatestSnapshot(
 		cacheP = persistedCache.put(snapshotKey, valueWithEpoch);
 		return cacheP;
 	};
+	const groupIds = useGroupIdsForSnapshotFetch ? [defaultGroupIdForSnapshot] : [];
```

Comment:
I worry it's too easy to make mistakes. I.e. some other layer (code above I looked at) assumes there is no difference between [] & undefined (i.e. defaultGroupIdForSnapshot would be added in both cases) . Vs. this code assumes there is a difference (as [] assumes defaultGroupIdForSnapshot should not be added). I think it's a bug

Reply:
One thing to consider - how do we reduce complexity?
While it reduces generality, I'd suggest doing the following:
1. get rid of array - almost pass a string.
2. Make it non-optional (with exception of driver entry point maybe)
3. invent some psudo-value that means "full snapshot" and use it across layers where needed (like prefetch flow), dropping / replacing with proper thing right before fetch().

Reply:
Will add a test for prefetch but I would still like to take an array from runtime to driver and then driver can care about how to convert that to string to pass it to service. Runtime should not care about it.